src/rpm/signature/pgp.rs

```diff
@@ -46,7 +46,8 @@ impl traits::Signing<traits::algorithm::RSA> for Signer {
                 ::pgp::packet::Subpacket::Issuer(self.secret_key.key_id()),
                 //::pgp::packet::Subpacket::SignersUserID("rpm"), TODO this would be a nice addition
             ]) // must be initialized
-            .build()?;
+            .build()
+            .map_err(|e| RPMError::SignError(Box::new(RPMError::Other(e))))?;
```

Comment:
Can we have a `RPMError::Signing(e: sd::error::Error)` instead?

Reply:
I thought so too but it is not as easy since `e` in this case is just a `String`. Thats one of the only reasons I introduced  the general `Other` error.
The alternative would be to create a custom (private) error type within the `pgp` crate to wrap this error string to make it a proper error.
Do you see any other solution?